Change to the python exporter

gabrielcocenza opened the pull request:
- remove the kibana exporter to use the newly created python exporter
- fixed / adapted the CI to use the new exporter

gabrielcocenza committed:
change to the python exporter

gabrielcocenza committed:
- install snapd that is not installed by default on ubuntu-latest
- set env vars before running the exporter binary

gabrielcocenza committed:
- fix CI to read the content of the metrics

Mehdi-Bendriss commented on .github/workflows/ci.yaml:
```diff
@@ -62,9 +62,11 @@ jobs:
           fetch-depth: 0
 
-      - name: Upgrade snapd
+      - name: Install snapd
```
Snapd comes pre-installed in ubuntu, so the initial aim was to refresh, i believe what we had previously was right

gabrielcocenza replied:
Initially I thought the same, but it seems that now the ubuntu-latest is not with snap installed:

See this run for example: https://github.com/canonical/opensearch-dashboards-snap/actions/runs/12869978875/job/35880547783

Mehdi-Bendriss reviewed (commented):
Thanks Gabriel, looks good! i have one small comment about snapd

gabrielcocenza committed:
- removed installation of yq. It's installed by default
- removed snap installation
- removed OpenSearch manual connections. OpenSearch has automatic
  connections.

gabrielcocenza committed:
- remove snapd refresh

phvalguima commented on .github/workflows/ci.yaml:
```diff
@@ -75,15 +66,6 @@ jobs:
-        run: |
-          sudo snap connect opensearch:log-observe
-          sudo snap connect opensearch:mount-observe
-          sudo snap connect opensearch:process-control
```
You still need the `process-control`.

phvalguima commented on .github/workflows/ci.yaml:
```diff
@@ -183,35 +166,36 @@ jobs:
-          if [ "$exporter_ok" ]; then
+            --retry-connrefused)
+
+           if [[ "$metrics" ==  *"opensearch_dashboards_up 1.0"* ]]; then
```
Nice!

phvalguima reviewed (changes requested):
Overall it is okay for me. [There is one snap connect you still need](https://github.com/canonical/opensearch-snap).
